Library/DiscUtils.Core/System/HashSet.cs

```diff
@@ -31,40 +49,68 @@ public bool Add(T item)
             return true;
         }
 
+        /// <summary>
+        /// Removes all elements from a HashSet.
+        /// </summary>
         public void Clear()
         {
             _innerDictionary.Clear();
             _innerDictionary = new Dictionary<T, bool>();
         }
 
+        /// <summary>
+        /// Determines whether a HashSet contains the specified element.
+        /// </summary>
+        /// <param name="item">The element to locate in the HashSet.</param>
+        /// <returns>true if the HashSet contains the specified element; otherwise, false.</returns>
         public bool Contains(T item)
         {
             return _innerDictionary.ContainsKey(item);
         }
 
+        /// <summary>
+        /// Copies the elements of a HashSet to an array, starting at the specified array index.
+        /// </summary>
+        /// <param name="array">The one-dimensional array that is the destination of the elements copied from
+        /// the HashSet. The array must have zero-based indexing</param>
+        /// <param name="arrayIndex">The zero-based index in array at which copying begins.</param>
         public void CopyTo(T[] array, int arrayIndex)
         {
             _innerDictionary.Keys.CopyTo(array, arrayIndex);
         }
 
+        /// <summary>
+        /// Gets the number of elements that are contained in a HashSet.
+        /// </summary>
         public int Count
         {
             get { return _innerDictionary.Keys.Count; }
         }
 
+        /// <summary>
+        /// Gets a value indicating whether the HashSet is read-only.
+        /// This property is always false.
+        /// </summary>
         public bool IsReadOnly
         {
-            get
-            {
-                return false;
-            }
+            get { return false; }
```

Comment:
Nuhuh.. This is not a comment or a warning suppression. Re-do the PR..

Just kidding.. This is great 👍